addons/website/static/src/snippets/s_instagram_page/000.js

```diff
@@ -1,81 +1,54 @@
+import { Interaction } from "@website/core/interaction";
+import { registry } from "@web/core/registry";
+
 import { _t } from "@web/core/l10n/translation";
-import publicWidget from "@web/legacy/js/public/public_widget";
-import { ObservingCookieWidgetMixin } from "@website/snippets/observing_cookie_mixin";
 
 // Note that Instagram can automatically detect the language of the user and
 // translate the embed.
 
-const InstagramPage = publicWidget.Widget.extend(ObservingCookieWidgetMixin, {
-    selector: ".s_instagram_page",
-    disabledInEditableMode: false,
+export class InstagramPage extends Interaction {
+    static selector = ".s_instagram_page";
+    disabledInEditableMode = false;
 
-    /**
-     * @override
-     */
     start() {
-        const iframeEl = document.createElement("iframe");
-        this.el.querySelector(".o_instagram_container").appendChild(iframeEl);
-        iframeEl.setAttribute("scrolling", "no");
-        iframeEl.setAttribute("aria-label", _t("Instagram"));
-        iframeEl.classList.add("w-100");
-        // We can already estimate the height of the iframe.
-        iframeEl.height = this._estimateIframeHeight();
         // We have to setup the message listener before setting the src, because
         // the iframe can send a message before this JS is fully loaded.
-        this.__onMessage = this._onMessage.bind(this);
-        window.addEventListener("message", this.__onMessage);
-
-        // We set the src now, we are ready to receive the message.
-        const src = `https://www.instagram.com/${this.el.dataset.instagramPage}/embed`;
-        this._manageIframeSrc(this.el, src);
+        this.addListener(window, "message", this.onMessage);
 
-        return this._super(...arguments);
-    },
-    /**
-     * @override
-     */
-    destroy() {
-        const iframeEl = this.el.querySelector(".o_instagram_container iframe");
-        if (iframeEl) {
-            iframeEl.remove();
-            window.removeEventListener("message", this.__onMessage);
-        }
-        this._super.apply(this, arguments);
-    },
+        this.renderIframe();
```

Comment:
Maybe render the iframe in the setup, and add the listener + set the src in `start()`